mapbox/services/uploads.py

```diff
@@ -57,6 +61,9 @@ def stage(self, filepath, creds=None):
             region_name="us-east-1")
 
         s3 = session.resource('s3')
+        if not os.path.exists(filepath):
+            raise validation.MapboxValidationError(
+                "{0} does not exist".format(filepath))
         with open(filepath, 'rb') as data:
```

Comment:
I was thinking about how we should raise `IOError` here instead of `MapboxValidationError`, and then realized that the thing to do is to take a file *object* here and not a filename. The existence check becomes none of the SDK's business. The CLI (and other clients) would take over this validation.